engine/crates/composition/src/emit_federated_graph.rs

```diff
@@ -20,7 +20,7 @@ pub(crate) fn emit_federated_graph(mut ir: CompositionIr, subgraphs: &Subgraphs)
         unions: mem::take(&mut ir.unions),
         scalars: mem::take(&mut ir.scalars),
         input_objects: mem::take(&mut ir.input_objects),
-        strings: mem::take(&mut ir.strings.strings),
+        strings: Vec::new(),
```

Comment:
This is another bug we just hadn't ran into: we were moving the contents of StringIr out, then kept using it in context, so duplicate string ids again.